### PR TITLE
Add StackParam JSON Unmarshaler

### PR DESCRIPTION
**Pull request by fyrchik**
### Problem

It is very convenient when StackParam is correctly unmarshaled from JSON.

### Solution

Implement custom UnmarshalJSON

### CODE REVIEW

File: pkg/rpc/stack_param.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+	if err = json.Unmarshal(data, &r); err != nil {
+		return
+	}
+
+	switch p.Type = r.Type; r.Type {
+	case ByteArray:
+		if err = json.Unmarshal(r.Value, &s); err != nil {
+			return
+		}
+		if b, err = hex.DecodeString(s); err != nil {
```
Comment (im-kulikov) [inline pkg/rpc/stack_param.go]:
```go
if p.Value, err = hex.DecodeString(s); err != nil {
```

File: pkg/rpc/stack_param.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+			return
+		}
+		p.Value = b
+	case String:
+		if err = json.Unmarshal(r.Value, &s); err != nil {
```
Comment (im-kulikov) [inline pkg/rpc/stack_param.go]:
```go
if err = json.Unmarshal(r.Value, &p.Value); err != nil {
```

File: pkg/rpc/stack_param.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+	case Integer:
+		if err = json.Unmarshal(r.Value, &i); err == nil {
+			p.Value = i
+			return
+		}
+		// sometimes integer comes as string
+		if err = json.Unmarshal(r.Value, &s); err != nil {
+			return
+		}
+		if i, err = strconv.ParseInt(s, 10, 64); err != nil {
```
Comment (im-kulikov) [inline pkg/rpc/stack_param.go]:
```go
if p.Value, err = strconv.ParseInt(s, 10, 64); err != nil {
```

**Comment by im-kulikov** [comment]
@anthdm thx

**Comment by hal0x2328** [comment]
Please reward fyrchik 300 points.

**Comment by lllwvlvwlll** [comment]
@fyrchik please provide a neo address for awards.

**Comment by lllwvlvwlll** [comment]
https://neoscan.io/transaction/2ce8e8bee250bf869670d40fe36d3cdba5a71da48f9c8dd777ac5734a921b8c0

**Pull request merged**